test/qr_test.go

```diff
@@ -0,0 +1,64 @@
+package test
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/xlab-si/emmy/client"
+	"github.com/xlab-si/emmy/common"
+	"github.com/xlab-si/emmy/config"
+	"github.com/xlab-si/emmy/log"
+	"math/big"
+	"testing"
+)
+
+func TestQRProof(t *testing.T) {
+	log.TurnOff() // otherwise too many communication messages
+	dlog := config.LoadDLog("pseudonymsys")
+	y1 := common.GetRandomInt(dlog.P)
+
+	qrClient, err := client.NewQRClient(testGrpcClientConn, dlog, y1)
+	if err != nil {
+		t.Errorf("Error when initializing NewQRClient")
+	}
+
+	proved, err := qrClient.Run()
+	if err != nil {
+		t.Errorf("Error when proving y is QR")
+	}
+
+	assert.Equal(t, proved, true, "QR proof does not work correctly")
+}
+
+func TestQNRProof(t *testing.T) {
+	log.TurnOff() // otherwise too many communication messages
+	qr := config.LoadQR("qrsmall")
+
+	// find some QNR:
+	/*
+		var y *big.Int
+		for {
+			y = common.GetRandomInt(qr.N)
+			isQR, err := qr.IsQR(y)
+			if err != nil {
+				continue
+			}
+			if !isQR {
+				fmt.Println("found-------------------")
+				fmt.Println(y)
+				break
+			}
+		}
+	*/
+	y, _ := new(big.Int).SetString("12104178836609367680573806127379667907452906068454411069540554013287299560619180831355030260792398116234585889489580787593876656468666620239704236202828625502145041222925221901747074982936094534863675673392263672910937547483238701395223837362913804135100562910313510524388491518118503153440479519531614792845582743827952578371105856240886035300256188119597496494955532842085616018276731618147827132295654573847443973865791627336995666490060797108039052526091724109438973877494599020695354285996982775389058505616271009101591441286538176405813622092518363027538767195421845003207037113604997900115835538908295151715618", 10)
```

Comment:
Maybe we should introduce a YAML file `testdata.yml` which we can put in `test/testdata` folder. In future :slightly_smiling_face: